src/types/list.rs

```diff
@@ -750,6 +849,32 @@ impl<'py> Iterator for BoundListIterator<'py> {
             None
         })
     }
+
+    #[inline]
+    #[cfg(all(not(Py_LIMITED_API), feature = "nightly"))]
```

Comment:
```suggestion
    #[cfg(feature = "nightly")]
```

Unless I'm missing something...

Reply:
I think you can delete the `#[cfg(not(Py_LIMITED_API))]` on `BoundListIterator::with_critical_section`, it was only there to avoid a clippy lint about unused code. `pyo3::sync::with_critical_section` is unconditionally exposed in the PyO3 API, it's just a no-op on GIL-enabled builds. We wanted to allow people to write code that avoids conditional compilation, where possible.